spec.md

```diff
@@ -2,7 +2,140 @@
 
 ### Overview
 
-### Goals
+### Goals and Requirements
+
+This storage specification is intended to support the following goals and 
+requirements.
+
+#### Local-first and Offline capable storage
+
+Users and apps need to be able to use (provision, set up, and start reading and 
+writing to) storage spaces without being connected to the internet.
+
+#### Storage and sharing of public, permissioned, and private encrypted data
+
+Although the local-first offline functionality is necessary, writing data to 
```

Comment:
Since this is more important, let's just lead with it instead of the less important thing?